app/src/main/java/com/douglasbruce/tasky/core/designsystem/component/AgendaReminder.kt

```diff
@@ -5,68 +5,184 @@ import androidx.compose.animation.fadeIn
+                },
             )
-        },
-        trailingContent = {
-            AnimatedVisibility(
-                visible = !isReadOnly,
-                enter = fadeIn(),
-                exit = fadeOut(),
-            ) {
-                Icon(
-                    imageVector = TaskyIcons.ChevronRight,
-                    contentDescription = null,
-                    tint = Black,
+            TaskyDropdownMenuItem(
+                text = stringResource(R.string.one_hour_before),
+                onClick = {
+                    agendaReminderState.selectedNotificationType.value = NotificationType.ONE_HOUR
+                    agendaReminderState.expanded.value = false
+                    onSelectionClick()
+                },
+            )
+            TaskyDropdownMenuItem(
+                text = stringResource(R.string.six_hours_before),
+                onClick = {
+                    agendaReminderState.selectedNotificationType.value = NotificationType.SIX_HOURS
+                    agendaReminderState.expanded.value = false
+                    onSelectionClick()
+                },
+            )
+            TaskyDropdownMenuItem(
+                text = stringResource(R.string.one_day_before),
+                onClick = {
+                    agendaReminderState.selectedNotificationType.value = NotificationType.ONE_DAY
+                    agendaReminderState.expanded.value = false
+                    onSelectionClick()
+                },
+            )
+        }
+    }
+}
+
+@Composable
+@ExperimentalMaterial3Api
+fun rememberAgendaReminderState(
+    initialSelectedNotificationType: NotificationType = NotificationType.THIRTY_MINUTES,
+    initiallyExpanded: Boolean = false,
+): AgendaReminderState = rememberSaveable(
+    saver = AgendaReminderState.Saver()
+) {
+    AgendaReminderState(
+        initialSelectedNotificationType = initialSelectedNotificationType,
+        initiallyExpanded = initiallyExpanded,
+    )
+}
+
+@Stable
+class AgendaReminderState constructor(
+    initialSelectedNotificationType: NotificationType = NotificationType.THIRTY_MINUTES,
+    initiallyExpanded: Boolean = false
+) {
+
+    var selectedNotificationType = mutableStateOf(NotificationType.THIRTY_MINUTES)
+    var expanded = mutableStateOf(false)
+
+    init {
+        selectedNotificationType.value = initialSelectedNotificationType
+        expanded.value = initiallyExpanded
+    }
+
+    companion object {
+        fun Saver(): Saver<AgendaReminderState, Any> = listSaver(
+            save = {
+                listOf(
+                    it.selectedNotificationType.value,
+                    it.expanded.value
+                )
+            },
+            restore = { value ->
+                AgendaReminderState(
+                    initialSelectedNotificationType = value[0] as NotificationType,
+                    initiallyExpanded = value[1] as Boolean
                 )
             }
```

Comment:
What made you decide to save this state in the composable itself rather than a ViewModel as usual?